client/app/components/empty-state/EmptyState.jsx

```diff
@@ -17,71 +17,64 @@ function createDashboard() {
   });
 }
 
-function getSteps({
-  showAlertStep,
-  showDashboardStep,
-  showInviteStep,
-}) {
-  const {
-    data_sources: dataSourceCount,
-    queries: queriesCount,
-    alerts: alertsCount,
-    dashboards: dashboardsCount,
-    users: usersCount,
-  } = organizationStatus.objectCounters;
-
-  return {
-    dataSource: {
-      show: true,
-      completed: dataSourceCount > 0,
-      render() {
-        return currentUser.isAdmin ?
-          (<span><a href="data_sources/new">Connect</a> a Data Source</span>) :
-          (<span>Ask an account admin to connect a data source.</span>);
-      },
-    },
-    query: {
-      show: true,
-      completed: queriesCount > 0,
-      render() {
-        return (<span><a href="queries/new">Create</a> your first Query</span>);
-      },
-    },
-    alert: {
-      show: showAlertStep,
-      completed: alertsCount > 0,
-      render() {
-        return (<span><a href="alerts/new">Create</a> your first Alert</span>);
-      },
-    },
-    dashboard: {
-      show: showDashboardStep,
-      completed: dashboardsCount > 0,
-      render() {
-        return (<span><a onClick={createDashboard}>Create</a> your first Dashboard</span>);
-      },
-    },
-    inviteUsers: {
-      show: showInviteStep,
-      completed: usersCount > 1,
-      render() {
-        return (<span><a href="users/new">Invite</a> your team members</span>);
-      },
-    },
-  };
+function Step({ show, completed, text, url, urlText, onClick }) {
+  return show ? (
+    <li className={classNames({ done: completed })}>
+      {onClick && urlText && <a href="javascript:void(0)" onClick={onClick}>{urlText}</a>}
+      {url && urlText && <a href={url}>{urlText}</a>}
+      {text && urlText ? ' ' : ''}
+      {text}
+    </li>
+  ) : null;
 }
 
+Step.propTypes = {
+  show: PropTypes.bool,
+  completed: PropTypes.bool,
+  text: PropTypes.string,
+  url: PropTypes.string,
+  urlText: PropTypes.string,
+  onClick: PropTypes.func,
+};
```

Comment:
You sure none of these should be required?

Reply:
Yep. Probably just `text`, but even `text` may be optional (why not?)

Reply:
Because a step without text makes no sense. :) I updated some props to be required (here and in EmptyState itself) in 78df7e7cc9f66eab8de97505371083a29ea0ab92.